### DISCUSSION TITLE
3D Multi-channel Signed Distance Fields ?

### DISCUSSION THREAD

**Discussion by JulienGery** [Chlumsky/msdfgen#217] [General]
Hello Mr. Chlumsky,
I hope you are doing well.

I have an idea, and I would like to get your opinion on it. I assume you are already familiar with raymarching. The idea would be to build a 3D texture representing the SDF of a 3D model. Such a process already exists, but it leaves much to be desired.

My idea is to explore whether it would be possible to "generalize" the multi-channel signed distance fields from your thesis to three dimensions to better preserve edges. Do you think this would be feasible?

**Comment by Chlumsky** [comment]
I am aware of using 3D distance fields represented by functions that are evaluated on-demand, but do people really use 3D textures to store them? That doesn't seem very efficient. But if so, then yes, it would be possible.

**Reply by JulienGery** [reply]
The only example i found where they use a signed distance field rasterized is the PolygonJS example. It also seems inefficient to me, but it has the advantage of being very general and computationally inexpensive once the rendering is done, even though the image is enormous.
